File: Cite.EvalIt.WebApp/src/app/core/query/tag.lookup.ts
import { IsActive } from '@app/core/enum/is-active.enum';
import { TagAppliesTo } from '@app/core/enum/tag-applies-to.enum';
import { TagType } from '@app/core/enum/tag-type.enum';
import { Lookup } from '@common/model/lookup';
import { Guid } from '@common/types/guid';

export class TagLookup extends Lookup implements TagFilter {
	ids: Guid[];
	excludedIds: Guid[];
	userIds: Guid[];
	excludedUserIds: Guid[];
	dataObjectIds: Guid[];
	excludedDataObjectIds: Guid[];
	isActive: IsActive[];
	appliesTo: TagAppliesTo[];
	type: TagType[];
	like: string;

	constructor() {
		super();
	}
}

export interface TagFilter {
	ids: Guid[];
	excludedIds: Guid[];
	userIds: Guid[];
	excludedUserIds: Guid[];
	dataObjectIds: Guid[];
	excludedDataObjectIds: Guid[];
	isActive: IsActive[];
	appliesTo: TagAppliesTo[];
	type: TagType[];
	like: string;
}

File: Cite.EvalIt/Query/TagLookup.cs
using Cite.EvalIt.Common;
using Cite.Tools.Data.Query;
using System;
using System.Collections.Generic;
using System.Text;

namespace Cite.EvalIt.Query
{
	public class TagLookup : Lookup
	{
		public String Like { get; set; }
		public List<Guid> Ids { get; set; }
		public List<Guid> ExcludedIds { get; set; }
		public List<Guid> UserIds { get; set; }
		public List<Guid> ExcludedUserIds { get; set; }
		public List<Guid> DataObjectIds { get; set; }
		public List<Guid> ExcludedDataObjectIds { get; set; }
		public List<IsActive> IsActive { get; set; }
		public List<TagAppliesTo> AppliesTo { get; set; }
		public List<TagType> Type { get; set; }
	}
}
